src/node_report.cc

```diff
@@ -299,6 +308,65 @@ void SetLoadTime() {
   localtime_r(&time_val.tv_sec, &loadtime_tm_struct);
 #endif
 }
+
+/*******************************************************************************
+ * Function to save the process command line
+ *******************************************************************************/
+void SetCommandLine() {
+  #ifdef __linux__
+    // Read the command line from /proc/self/cmdline
+    char buf[64];
+    FILE* cmdline_fd = fopen("/proc/self/cmdline", "r");
+    if (cmdline_fd == nullptr) {
+      return;
+    }
+    commandline_string = "";
+    int bytesread = fread(buf, 1, sizeof(buf), cmdline_fd);
+    while (bytesread > 0) {
+      for (int i = 0; i < bytesread; i++) {
+        // Arguments are null separated.
+        if (buf[i] == '\0') {
+          commandline_string += " ";
+        } else {
+          commandline_string += buf[i];
+        }
+      }
+      bytesread = fread(buf, 1, sizeof(buf), cmdline_fd);
+    }
+    fclose(cmdline_fd);
+  #elif __APPLE__
+    char **argv = *_NSGetArgv();
+    int argc = *_NSGetArgc();
+
+    commandline_string = "";
+    std::string seperator = "";
+    for (int i = 0; i < argc; i++) {
+      commandline_string += seperator + argv[i];
+      seperator = " ";
+    }
+#elif _AIX
```

Comment:
The rest of the function was wrong and didn't match the file. I'd indented when starting #ifdef sections. I've adjusted the rest of the function to be correct.